src/datasources/locking-api/locking-api.service.ts

```diff
@@ -24,6 +25,27 @@ export class LockingApi implements ILockingApi {
       this.configurationService.getOrThrow<string>('locking.baseUri');
   }
 
+  async getCampaigns(args: {
+    limit?: number;
+    offset?: number;
+  }): Promise<Page<Campaign>> {
+    try {
+      const url = `${this.baseUri}/api/v1/campaigns`;
+      const { data } = await this.networkService.get<Page<Campaign>>({
```

Comment:
Do we know how often the campaign data will update/whether it's cached by the Locking Service? It might be beneficial for us to cache it. What do you think?

Reply:
After discussion, it has been confirmed that the Locking Service will cache. We will leave this as is for now.